client/jetpack-connect/woo-dna-config.js

```diff
@@ -1,13 +1,30 @@
 /**
- * Internal dependencies
+ * This function abstracts the logic for the "Woo DNA" Jetpack flow. The Woo DNA flow is an exclusively Woo-branded
+ * Jetpack connection flow. It's meant to be used by Woo extensions or services with the Jetpack Connection package.
+ * Because the user may have never heard of Jetpack, and hasn't installed Jetpack in their site, it's important to
+ * not fill the flow with potentially confusing Jetpack branding.
+ *
+ * @param query {Object} Map with the querystring parameters of the current page.
+ * @returns An object with helper functions that are documented below.
  */
-import { isEnabled } from 'config';
+export default ( query ) => ( {
```

Comment:
I like that factory function! 👍

Reply:
How about wrapping it with `Object.freeze` to prevent mutations?

Reply:
It doesn't hurt :)

Done in https://github.com/Automattic/wp-calypso/pull/43282/commits/eb9637190087f6767fe53f857282bb8fc6550c13 . I just added `Object.freeze`, the change looks bigger because `prettier` did its thing.